pipelines/api-pipelines.yml

```diff
@@ -0,0 +1,97 @@
+trigger:
+  branches:
+    include:
+      - master
+  paths:
+    include:
+      - api/*
+      - pipelines/api-pipelines.yml
+  
+pr:
+  autoCancel: true   
+  branches:
+    include:
+     - master
+  paths:
+    include:
+      - api/*
+      - pipelines/api-pipelines.yml
+
+pool:
+  vmImage: "ubuntu-latest"
+  
+variables:
+  azureSubscription: PELITTLE TEAM - CSE DWR(d36d0808-a967-4f73-9fdc-32ea232fc81d)
+
+stages:
+  - stage: BuildStage
+    displayName: Build API
+    jobs:
+      - job: BuildJob
+        displayName: Build Test and Publish
+        steps:
+          - task: UseDotNet@2
+            displayName: Dotnet version 3.1.100
+            inputs:
+              version: '3.1.100'
+
+          - task: DotNetCoreCLI@2
+            displayName: Build
+            inputs:
+              command: build
+              projects: '**/*.csproj'
+              arguments: '--configuration Release'
+
+          - task: DotNetCoreCLI@2
+            displayName: Test
+            inputs:
+              command: test
+              projects: '**/*Tests/*.csproj'
+              arguments: '--configuration $(buildConfiguration) --collect "Code coverage"'
+  
+          - task: DotNetCoreCLI@2
+            displayName: Publish packages
+            inputs:
+              command: publish
+              arguments: '--configuration $(BuildConfiguration) --output $(Build.ArtifactStagingDirectory)'
+              zipAfterPublish: True
+  
+          - task: PublishBuildArtifacts@1
+            condition: and(succeeded(), ne(variables['Build.Reason'], 'PullRequest'))
+            displayName: Publish Artifacts
+            inputs:
+              pathtoPublish: '$(Build.ArtifactStagingDirectory)' 
+              artifactName: 'drop'
+              publishLocation: 'Container'
+             
+  - stage: DeployStage
+    condition: and(succeeded(), ne(variables['Build.Reason'], 'PullRequest'))
```

Comment:
This condition will ensure to not run this stage if this pipeline run is triggered via PR. This stage will only deploy the api changes when the PR is merged to master.

Reply:
Nice! Good to know you can specify conditions for this.  Let's discuss whether this is the way to go vs. splitting this template out into 2 pipelines (1 for build/validation (CI) and the other for publishing (CD)).  In the past i've typically split them out to keep them easier to manage & understand.

Reply:
I initially thought about splitting the pipeline into two but then also thought to keep everything for api into one place. I`m completely flexible to break it down into two if that is more manageable & readable. Lets discuss this after the planning today and then I will make the changes accordingly.

Reply:
I would recommend using templates to make common tasks/stage/jobs reusable

Reply:
Discussed this offline with Wallace & Jafar, for now we are keeping the approach to build and deploy within the same pipeline. We will consider using templates later when we have lots of reusable pieces.